packages/cf-component-icon/src/Icon.js

```diff
@@ -1,33 +1,245 @@
+      fontPoint = "'\\F014'";
+      break;
+    case 'plus':
+      fontPoint = "'\\F015'";
+      break;
+    case 'happy':
+      fontPoint = "'\\F016'";
+      break;
+    case 'sad':
+      fontPoint = "'\\F017'";
+      break;
+    case 'search':
+      fontPoint = "'\\F018'";
+      break;
+    case 'wrench':
+      fontPoint = "'\\F019'";
+      break;
+    case 'remove-sign':
+    case 'remove':
+      fontPoint = "'\\F01A'";
+      break;
+    case 'speech':
+      fontPoint = "'\\F01B'";
+      break;
+    case 'gear':
+      fontPoint = "'\\F01C'";
+      break;
+    case 'help':
+      fontPoint = "'\\F01D'";
+      break;
+    case 'widen':
+      fontPoint = "'\\F022'";
+      break;
+    case 'calendar':
+      fontPoint = "'\\F01E'";
+      break;
+    case 'facebook':
+      fontPoint = "'\\F01F'";
+      break;
+    case 'twitter':
+      fontPoint = "'\\F020'";
+      break;
+    case 'googleplus':
+      fontPoint = "'\\F021'";
+      break;
+    case 'linkedin':
+      fontPoint = "'\\F022'";
+      break;
+    case 'resize-horizontal':
+      fontPoint = "'\\F024'";
+      break;
+    case 'reorder':
+    case 'resize-vertical':
+      fontPoint = "'\\F024'";
+      additionalRules.transform = 'rotate(90deg)';
+      break;
+    case 'upload':
+      fontPoint = "'\\F025'";
+      break;
+    case 'close':
+      fontPoint = "'\\D7'";
+      additionalRules.fontFamily = 'Open Sans, Helvetica, Arial, sans-serif';
+      break;
+  }
+
+  if (border) {
+    additionalRules.border = `1px solid ${color}`;
+    additionalRules.padding = '.2em .25em .15em';
+    additionalRules.borderRadius = '2px';
+  }
 
-    if (this.props.size) {
-      className += ` cf-icon--${this.props.size}`;
+  if (spin) {
+    additionalRules.display = 'inline-block';
+    additionalRules.animation = `${theme.spinAnimationName} 2s infinite linear`;
+  }
+
+  return {
+    color,
+    fontSize,
+    fontStyle: 'normal',
+
+    fontFamily: 'Cloudflare Icons',
+
+    '::before': {
+      ...additionalRules,
+      content: fontPoint
     }
+  };
+}, 'i');
 
-    if (this.props.border) className += ' cf-icon--border';
-    if (this.props.spin) className += ' cf-icon--spin';
-    if (this.props.muted) className += ' cf-icon--muted';
-    if (this.props.white) className += ' cf-icon--white';
+const Label = createComponent(() => ({
+  display: 'none'
+}));
 
+class Icon extends React.Component {
+  render() {
     return (
-      <i className={className} role={this.props.role}>
-        {this.props.label &&
-          <span className="cf-icon__label">{this.props.label}</span>}
-      </i>
+      <Wrapper {...this.props}>
+        <Label>{this.props.label}</Label>
+      </Wrapper>
     );
   }
 }
 
 Icon.propTypes = {
-  label: PropTypes.oneOfType([
-    PropTypes.string,
-    PropTypes.oneOf([false])
-  ]).isRequired,
+  /**
+   * The a11y label set for screen-readers(?). Needs to correspond with the type
+   * prop.
+   */
+  label: PropTypes.oneOf([
```

Comment:
Cool! I wanted that too but was worried that label needed to be something specific. Since no one has made any additional comments I think we should def do that. Will change this.